### ISSUE TITLE
❌ Build Error: An exception occurred in module resolution.

### ISSUE THREAD

**Issue by Y80** [vercel/next.js#69174]
### Link to the code that reproduces this issue

https://stackblitz.com/edit/stackblitz-starters-guwhtn?file=app%2Factions.ts

### To Reproduce

1. Open the demo link and wait for the dependencies installation to complete.
2. Execute the command `npm run build`.
3. An exception has occurred.

### Current vs. Expected behavior

Build error info: 
`Cannot get final name for export 'encodeXML' of ./node_modules/entities/lib/esm/index.js`

😄 Related issue: #58643 

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: Ubuntu 20.04.0 LTS Thu Aug 22 2024 16:33:23
  Available memory (MB): NaN
  Available CPU cores: 8
Binaries:
  Node: 18.20.3
  npm: 10.2.3
  Yarn: 1.22.19
  pnpm: 8.15.6
Relevant Packages:
  next: 14.2.6 // Latest available version is detected (14.2.6).
  eslint-config-next: 13.5.1
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Module Resolution

### Which stage(s) are affected? (Select all that apply)

next build (local)

### Additional context

When run `next dev`, there are no issues. When run `next build`, the issues occurred.

In the demo, cheerio@1.0.0 is used, whice depends on entities@4.5.0.

**If I use cheerio@1.0.0-rc.10 or earlier version,  whice depends on earlier version of entities, and the build process can be executed normally.** ✅

In the code, I imported an **action** (from `app/actions.ts` ) in a client side component, and the `app/actions.ts` file imported another module which used cheerio. If these code related to cheerio are removed, the problem will disappear.

**Comment by huozhi** [comment]
Thanks fo reporting. This issue is already fixed on Next.js canary

**Comment by huozhi** [comment]
Hi @Y80 , the fix is back ported to 14.2.8, please upgrade!
